### PR TITLE
tree: Strong TreeNode type

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Make TreeNode get nominal typing, and clean up tests this breaks.

## Breaking Changes

TreeNodes now have nominal typing: in some cases this will require changes like constructing unhydrated nodes to make field assignment compile.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/api-report/tree.api.md [hunk 1]
```diff
@@ -25,9 +25,11 @@ import type { TSchema } from '@sinclair/typebox';
 // @internal
 export function adaptEnum<TScope extends string, const TEnum extends Record<string, string>>(factory: SchemaFactory<TScope>, members: TEnum): (<TValue extends TEnum[keyof TEnum]>(value: TValue) => {
     readonly value: TValue;
+    readonly "__#124291@#brand": unknown;
```
Comment (CraigMacomber) [inline packages/dds/tree/api-report/tree.api.md:28]: the TypeScript compiler seems to generate this number somehow as part of its handling of JS private fields. It seems to only show up like this in some edge case which is used in the enum helpers. I wonder if I can refactor them to make it go away.

Comment (CraigMacomber) [reply packages/dds/tree/api-report/tree.api.md:28]: I found a fix.

**Review by DLehenbauer — APPROVED** [review]
Approved, but I would appreciate it if you would leave some coverage for the 'const foo = _.object(..)' style declarations.

Perhaps keep the updates to 'objectFactory.spec.ts' and restore 'object.spec.ts'?  ('object.spec.ts' is focused on verifying that '_.object(..)' creates nodes that are indistinguishable from object literals ('{}'), which is why it has so many skipped tests at the moment.)

**Comment by CraigMacomber** [comment]
> Perhaps keep the updates to 'objectFactory.spec.ts' and restore 'object.spec.ts'? ('object.spec.ts' is focused on verifying that '_.object(..)' creates nodes that are indistinguishable from object literals ('{}'), which is why it has so many skipped tests at the moment.)

Done.

**Pull request merged**